source/_docs/configuration/templating.markdown

```diff
@@ -72,7 +72,7 @@ Not supported in [limited templates](#limited-templates).
 - Iterating `states` will yield each state sorted alphabetically by entity ID.
 - Iterating `states.domain` will yield each state of that domain sorted alphabetically by entity ID.
 - `states.sensor.temperature` returns the state object for `sensor.temperature` (avoid when possible, see note below).
-- `states('device_tracker.paulus')` will return the state string (not the object) of the given entity or `unknown` if it doesn't exist.
+- `states('device_tracker.paulus')` will return the state string (not the object) of the given entity, `unknown` if it doesn't exist, `unavailable` if the object exists but not yet available (e.g. due to an integration internal processing) 
```

Comment:
What does "due to an integration internal processing" mean?

Reply:
Hello,
The root cause of this PR is because, I've got some error in templating with error such as 'None Type'.
When I look deeper in HomeAssistant source code, I've seen that for templating, I got "3 states" : 
- `STATE_UNKNOWN` (const.py)
- `STATE_UNAVAILABLE` (const.py)
- the native state value

Unknown was quite clear (e.g. 'none')
Unavailable can be set by the integration via the method available (or _attr_available attribute) (cf. `helpers/entity.py`)
That's why I put  

> integration internal processing

If you have another proposal, I take

Reply:
My proposal would be:

```suggestion
- `states('device_tracker.paulus')` will return the state string (not the object) of the given entity, `unknown` if it doesn't exist, `unavailable` if the object exists but is not yet available.
```

Reasoning: I have no idea what internal processing means. `unavailable` has nothing to do with Home Assistant internals.

Reply:
I upate